### PR TITLE
 Bug 1151843 - [Messages][Refactoring] Use notification "data" attribute instead of icon URL query string parameters. r=julien

### PR DESCRIPTION
**Pull request by azasypkin**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=937d1aff700ed4f8b35f8fb722436c3d620ed44b

File: apps/sms/js/activity_handler.js [hunk 1]
```diff
@@ -400,9 +390,10 @@ var ActivityHandler = {
           }
 
           var options = {
-            icon: iconURL,
+            icon: NotificationHelper.getIconURI(app),
             body: body,
-            tag: 'threadId:' + threadId
+            tag: 'threadId:' + threadId,
+            data: { id, threadId }
```
Comment (azasypkin) [inline apps/sms/js/activity_handler.js]: I don't see that we use ```number``` parameter, so removed it.

File: apps/sms/js/activity_handler.js [hunk 2]
```diff
@@ -518,28 +500,20 @@ var ActivityHandler = {
...
 
-    navigator.mozApps.getSelf().onsuccess = function(event) {
-      var app = event.target.result;
-
-      app.launch();
+     navigator.mozApps.getSelf().then((app) => {
+       app.launch();
 
-      // the type param is only set for class0 messages
-      if (params.type === 'class0') {
```
Comment (azasypkin) [inline apps/sms/js/activity_handler.js]: Looks like after the patch for the bug [1] we can't have notification for class-0 messages :)

[1] https://bugzilla.mozilla.org/show_bug.cgi?id=858908

File: shared/test/unit/mocks/mock_navigator_moz_apps.js [hunk 3]
```diff
@@ -41,6 +34,53 @@ var MockNavigatormozApps = {
     }
   },
 
+  mCreateDOMRequest: function() {
```
Comment (azasypkin) [inline shared/test/unit/mocks/mock_navigator_moz_apps.js]: Maybe a bit overcomplicated, don't have strong opinion on that, ideally it should be in a separate mock file, but to use it every test should explicitly ```require``` it :(

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=48218b6b1993267e240dbc0fa73f95da37be726c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bc13b20bfd74aecd07cef336071068e6c60019bb

**Comment by julienw** [comment]
I just looked the 2nd commit for now. You'll need other peers reviews too so maybe it's best to separate it in its separate bug?

**Comment by azasypkin** [comment]
> I just looked the 2nd commit for now. You'll need other peers reviews too so maybe it's best to separate it in its separate bug?

Yep, that is why I asked your opinion on that :) Will create dedicated refactoring bug for it!

File: shared/test/unit/mocks/mock_notification.js [hunk 4]
```diff
@@ -12,6 +12,7 @@ function MockNotification(title, options) {
   this.icon = options.icon || undefined;
   this.body = options.body || undefined;
   this.tag = options.tag || undefined;
+  this.data = options.data || undefined;
```
Comment (julienw) [inline shared/test/unit/mocks/mock_notification.js]: when I read this code, I think we should just remove all `|| undefined`... how about empty strings or `0` ?

Comment (azasypkin) [inline shared/test/unit/mocks/mock_notification.js]: I had the same feeling as well, will remove it and see what will happen :)

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bf674faee88ec267fb94b04bab4a5a29d3c11f77

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fdc17468ab50549bebefe02f7efdc412fa34dffe

**Pull request merged**